File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        int l = strs.size();
        map<string,vector<string>>mp;
        for(int i=0;i<l;i++){
            string s = strs[i];
            sort(s.begin(),s.end());
            if(mp.find(s) != mp.end()){
                mp[s].push_back(strs[i]);
            }
            else{
                mp[s] = {strs[i]};
            }
        }
        vector<vector<string>>res;
        map<string,vector<string>>::iterator itr;
        for(itr = mp.begin(); itr != mp.end(); itr++){
            res.push_back(itr->second);
        }
        return res;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        
        Map<String, List<String>> anagrams = new HashMap<>();
        
        for(String str : strs){
            char[] currentString = str.toCharArray();
            Arrays.sort(currentString);
            String sortedStr = new String(currentString);
            anagrams.computeIfAbsent(sortedStr, k -> new ArrayList<>()).add(str);
        }
        
        List<List<String>> res = new ArrayList<>();
        
        for(List<String> str : anagrams.values()){
            res.add(str);
        }
        
        return res;
    }
}